tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -622,24 +664,20 @@ func (az *AzureProvider) constructNetworkProfile(ctx context.Context,
 		return nil, fmt.Errorf("cannot obtain security group rules of worker subnet")
 	}
 
-	vmRandString := strings.Split(vmName, "-")[1]
-	var nsgName string
-	if !checkForNil(nsgStruct.ID) {
-		nsgID := *(nsgStruct.ID)
-		nsgName = extractResourceName(nsgID)
-		err := az.createSecurityGroupRules(ctx, nsgName)
-		if errorCheck(err) {
-			return nil, fmt.Errorf("failed to update security group rules: %v", err)
-		}
-	} else {
-		nsgName = az.generateResourceName("nsg", vmRandString)
-		err := az.createSecurityGroupRules(ctx, nsgName)
-		if errorCheck(err) {
-			return nil, fmt.Errorf("failed to update security group rules: %v", err)
-		}
+	if checkForNil(nsgStruct.ID) {
+		return nil, fmt.Errorf("failed to find worker NSG")
 	}
+
+	vmRandString := strings.Split(vmName, "-")[1]
```

Comment:
nit: it took me a while to figure out the purpose of vmRandString, `vmRandomID` or something similar could be more intuitive.

Reply:
This is part of existing code and renaming it would require changing parts of code that has nothing to do with this change.